apex/cli/addon_shipping_cli/sign_xpi.py

```diff
@@ -0,0 +1,78 @@
+"""
+The CLI command for the sign-xpi lambda.
+"""
+
+import argparse
+import boto3
+import hashlib
+import json
+import os.path
+import sys
+import traceback
+
+DEFAULT_S3_BUCKET = 'eglassercamp-addon-sign-xpi-input'
+
+s3 = boto3.resource('s3')
+aws_lambda = boto3.client('lambda')
+
+parser = argparse.ArgumentParser(description="Upload an XPI and cause it to be signed.")
+parser.add_argument('-t', '--type', help="Type of XPI (system or privileged)",
+                    choices=['system', 'privileged'], required=True)
+parser.add_argument('-s', '--s3-source', nargs='?',
+                    help='S3 bucket to upload XPI to for signing')
+parser.add_argument('xpi_file', type=file, help="Filename of XPI to sign")
+
+
+def main(args=sys.argv[1:]):
+    parameters = parser.parse_args(args)
+
+    xpi_file = parameters.xpi_file
+    xpi_sha256 = sha256(xpi_file)
+    xpi_file.seek(0)
+    bucket_name = parameters.s3_source or DEFAULT_S3_BUCKET
+    bucket = s3.Bucket(bucket_name)
+    key = os.path.basename(xpi_file.name)
+    bucket.put_object(Body=xpi_file, Key=key)
```

Comment:
Since this is only input to the signer, I think it's OK if we overwrite previous ones.